### PR TITLE
Add mjr endpoint

### PR DESCRIPTION
**Pull request by southeo**
- Get MJR record by job id
- Get paginated list of MJRs by creator (may filter by state (failed, scheduled, completed)

[JIRA](https://naturalis.atlassian.net/browse/DD-810?atlOrigin=eyJpIjoiZjExYTQ4OWRkMTUxNDZlZWIxMDhlN2M1N2I2NzQzOGIiLCJwIjoiaiJ9)

### CODE REVIEW

File: src/main/java/eu/dissco/backend/controller/MasJobRecordController.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@Slf4j
+@RestControllerAdvice
+@RestController
+@RequestMapping("api/v1/mjr")
```
Comment (samleeflang) [inline src/main/java/eu/dissco/backend/controller/MasJobRecordController.java:25]: The call I am missing is to get jobs called for a particular specimen.
Maybe this should be a call/operation on the specimen (specimens/XXX/XXX/mjr) returns all mjr's for that specimen. Maybe with a request param where you can add the status, so you can select only the ones that are scheduled or failed.
Probably best to make a new task for this.

File: src/main/java/eu/dissco/backend/controller/MasJobRecordController.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    this.service = service;
+  }
+
+  @GetMapping(value = "/{jobId}", produces = MediaType.APPLICATION_JSON_VALUE)
+  public ResponseEntity<JsonApiWrapper> getMasJobRecord(
+      @PathVariable("jobId") UUID jobId, HttpServletRequest request) throws NotFoundException {
+    return ResponseEntity.ok().body(service.getMasJobRecordById(jobId, getPath(request)));
+  }
+
+  @GetMapping(value = "/creator/"
```
Comment (samleeflang) [inline src/main/java/eu/dissco/backend/controller/MasJobRecordController.java:42]: Wonder if this is a call/operation on the user object and should therefore be in the user controller. What do you think?

File: src/main/java/eu/dissco/backend/service/MasJobRecordService.java [hunk 3]
```diff
@@ -16,15 +25,60 @@
...
+    if (state == null) {
+      masJobRecordsPlusOne = masJobRecordRepository.getMasJobRecordsByCreator(creatorId, pageNum,
+          pageSizeToCheckNext);
+    } else {
+      masJobRecordsPlusOne = masJobRecordRepository.getMasJobRecordsByCreatorAndStatus(creatorId,
+          state.getState(), pageNum, pageSizeToCheckNext);
+    }
+    boolean hasNext = masJobRecordsPlusOne.size() > pageSize;
+    var masJobRecords = hasNext ? masJobRecordsPlusOne.subList(0, pageSize) : masJobRecordsPlusOne;
+    List<JsonApiData> dataList = new ArrayList<>();
```
Comment (samleeflang) [inline src/main/java/eu/dissco/backend/service/MasJobRecordService.java:56]: Couldn't this be a stream?

**Review by samleeflang — COMMENTED** [review]
Nice addition! Looking forward seeing this in the FE

File: src/main/java/eu/dissco/backend/controller/MasJobRecordController.java (hunk 1 above)
Comment (southeo) [reply src/main/java/eu/dissco/backend/controller/MasJobRecordController.java:25]: [Added new task](https://naturalis.atlassian.net/browse/DD-834)

File: src/main/java/eu/dissco/backend/controller/MasJobRecordController.java (hunk 2 above)
Comment (southeo) [reply src/main/java/eu/dissco/backend/controller/MasJobRecordController.java:42]: "Creator" here is the MAS Id, not user that called the user that scheduled the MAS, so I think this endpoint is the best. 
We don't actually capture the user information in the Mas Job Record. I can a user might want to see all their scheduled jobs. Added a [new jira story](https://naturalis.atlassian.net/browse/DD-835) to modify this object

Comment (samleeflang) [reply src/main/java/eu/dissco/backend/controller/MasJobRecordController.java:42]: Oh, yes sorry that would be nice, let's add that info :+1:

**Review by samleeflang — DISMISSED** [review]
:clap:

**Review by samleeflang — APPROVED** [review]

**Pull request merged**